scss/components/_button.scss

```diff
@@ -161,10 +161,9 @@ $button-transition: background-color 0.25s ease-out, color 0.25s ease-out !defau
 @mixin button-disabled($color: $primary-color) {
   opacity: $button-opacity-disabled;
   cursor: not-allowed;
-
+  background-color: $color;
   &:hover, &:focus {
-    background-color: $color;
-    color: $button-color;
```

Comment:
The issue with #9617 is the text color is changed from the _a11y-compliant color_ to the default `$button-color` on mouse over, because `@mixin button-disabled` force the text color on `:hover` and `focus` with the wrong color.

Removing `color: $button-color;` fix this issue in the wrong way: it makes a disabled button not a11y-compliant at all. A better (but a bit repetitive) solution would be to use:
```sass
color-pick-contrast($background, ($button-color, $button-color-alt))
```
... like in `@mixin button-style`.